geopandas/geodataframe.py

```diff
@@ -193,6 +193,38 @@ def set_geometry(self, col, drop=False, inplace=False, crs=None):
         if not inplace:
             return frame
 
+    def rename_geometry(self, col,inplace=False):
+        """
+        Renames the GeoDataFrame geometry column to
+        the specified name. By default yields a new object.
+
+        The original geometry column is replaced with the input.
+
+        Parameters
+        ----------
+        col : new geometry column label
+        inplace : boolean, default False
+            Modify the GeoDataFrame in place (do not create a new object)
+
+        Examples
+        --------
+        >>> df1 = df.rename_geometry('geom1')
+        >>> df1.geometry.name
+        >>> geom1
```

Comment:
```suggestion
        'geom1'
```

Reply:
@martinfleis  Thank You for the code review. I have a rather silly question. I am getting a commit suggestion for each change, should I go ahead with it. The question is would it create multiple commits? or be squashed to a single commit? or it would be better to make the suggested changed locally and commit and push my code.
Regarding `col` attribute name, only thing I could think of would be `geo_col` or `geo_name`. Since it the new geometry name. Let me know about your thoughts and suggestions.
Regarding adding the user guide: I am new to `.rst` files. If it is not a lot to ask, I would need some help in this aspect. It would be great if you can point me in what direction to look at.

Reply:
Go ahead with commits, I will squash it to a single commit in the end anyway.

Keep `col` for now.

You can ignore user guide, just add 

    .. automethod:: geopandas.GeoDataFrame.rename_geometry

to `doc/source/reference.rst` to GeoDataFrame part. That's enough.